irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -729,37 +800,50 @@ namespace iroha {
           },
           [&](auto range, auto &) {
             if (range.empty()) {
-              return error_response<
-                  shared_model::interface::NoAssetErrorResponse>;
+              return query_response_factory_->createErrorQueryResponse(
+                  QueryErrorType::kNoAsset,
+                  "no asset " + q.assetId()
+                      + " in account with such id: " + creator_id_,
+                  query_hash_);
             }
 
-            return apply(range.front(), [&q](auto &domain_id, auto &precision) {
-              return QueryResponseBuilder().assetResponse(
-                  q.assetId(), domain_id, precision);
-            });
-          });
+            return apply(
+                range.front(), [this, &q](auto &domain_id, auto &precision) {
+                  auto asset = common_objects_factory_->createAsset(
+                      q.assetId(), domain_id, precision);
+                  return asset.match(
+                      [this](expected::Value<std::unique_ptr<
+                                 shared_model::interface::Asset>> &asset) {
+                        return query_response_factory_->createAssetResponse(
+                            std::move(asset.value), query_hash_);
+                      },
+                      [this](const expected::Error<std::string> &err) {
+                        log_->error("could not create asset: {}", err.error);
+                        return query_response_factory_
+                            ->createErrorQueryResponse(
+                                QueryErrorType::kStatefulFailed,
+                                "could not create asset object: " + err.error,
```

Comment:
the same error is duplicated for log and here, please move it to a common variable.